### ISSUE TITLE
Fix HDF5 1.8 incompatibilities

### ISSUE THREAD

**Issue by prudhomm** [feelpp/feelpp#1140]
HDF5 1.8 is not well supported, we get a lot of messages like this
```
HDF5-DIAG: Error detected in HDF5 (1.8.16) MPI-process 0:
  #000: ../../../src/H5FDmpio.c line 344 in H5Pset_fapl_mpio(): not a file access list
    major: Property lists
    minor: Inappropriate type
  #001: ../../../src/H5Pint.c line 3381 in H5P_object_verify(): property list is not a member of the class
    major: Property lists
    minor: Unable to register new atom
HDF5-DIAG: Error detected in HDF5 (1.8.16) MPI-process 0:
  #000: ../../../src/H5FDmpio.c line 485 in H5Pset_dxpl_mpio(): not a dxpl
    major: Property lists
    minor: Inappropriate type
  #001: ../../../src/H5Pint.c line 3381 in H5P_object_verify(): property list is not a member of the class
    major: Property lists
    minor: Unable to register new atom
  #002: ../../../src/H5Pint.c line 3331 in H5P_isa_class(): not a property list
    major: Invalid arguments to routine
    minor: Inappropriate type
```
when creating the feelpp partitioned hdf5 mesh (eg with `feelpp_qs_laplacian_*` 

/cc @vincentchabannes @Trophime

**Comment by prudhomm** [comment]
Should we move to 1.10 in docker to have a clean mpi+hdf5 behavior?

**Comment by Trophime** [comment]
To get hdf5 1.10 there are 2 routes:
* either recompile hdf5 and keep ubuntu Xenial
* move ubuntu Bionic

The second solution shall be the more "long term" one. However, right now, it's risky since Bionic based images won't be supported by Cesga!!

**Comment by prudhomm** [comment]
Bionic is already there but as you said we should stick to 16.04 for the time being. @vincentchabannes what do you think?

**Comment by vincentchabannes** [comment]
Don't forget that on bionic, it's openmpi 2 and there some issue with petsc. We need to recompile all

**Comment by prudhomm** [comment]
yes, there is that too. is openmpi 3.1.2 reliable because we should also move to a newer version to get rid of these `/proc/mounts` messages in docker. Or should we try another flavor of mpi _e.g._ mpich?

**Comment by prudhomm** [comment]
@vincentchabannes we need to take a decision, what were your findings in the end with openmpi3 , hdf5 and feelpp ?

**Comment by Trophime** [comment]
@prudhomm can we have a table with all feelpp requirements versions for a given version  (including also remarks on patched version of some libs contained in feelpp/contrib)? It will help to check what is missing on a distro base (ie what has to be packaged/installed)

**Comment by Trophime** [comment]
and also to note the cross-dependencies: eg eigen3/paraview; mpi4py/paraview; gmsh/metis on so ...

**Comment by prudhomm** [comment]
@trophime the requirements are in the docs. Check out the different versions

**Comment by Trophime** [comment]
sure but there is no mention of eigen3 and any of other "modified" libs used by feelpp...
Because this may be an issue (for instance try to build with paraview support)

**Comment by prudhomm** [comment]
@Trophime I build paraview for feelpp integration and never got these issues

**Comment by vincentchabannes** [comment]
@Trophime for paraview, we need typically to add to cmake configuration : 
```
    -DPARAVIEW_ENABLE_CATALYST=ON     -DPARAVIEW_INSTALL_DEVELOPMENT_FILES=ON 
```
in order to support paraview catalyst. I don't know if it's really tested with minimal install of paraview, probably we need

**Comment by prudhomm** [comment]
@Trophime I will document as you suggest the `contrib` libs.
regarding eigen3, we simply use a new version with features from `Eigen::Tensor`

**Comment by Trophime** [comment]
Ok great.
With latest eigen3 3.3.5 shipped with Debian/Testing it FTBS when using system eigen.
So you must use a "more" recent version?
